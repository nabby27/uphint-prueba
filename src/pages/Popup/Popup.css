.App {
  position: absolute;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  text-align: center;
  height: 100%;
  padding: 10px;
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.RecordingText {
  color: #61dafb;
  font-size: 20px;
  font-weight: bold;
  margin-top: 10px;
}

.RecordingText:after {
  overflow: hidden;
  display: inline-block;
  vertical-align: bottom;
  -webkit-animation: ellipsis steps(4, end) 900ms infinite;
  animation: ellipsis steps(4, end) 1000ms infinite;
  content: "\2026";
  width: 0px;
}

@keyframes ellipsis {
  to {
      width: 1.25em;
  }
}