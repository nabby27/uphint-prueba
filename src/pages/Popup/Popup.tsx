import { useEffect, useState } from 'react';
import { StateService } from '../../service/State.service';
import { Logo } from './Logo/Logo';
import './Popup.css';
import { RecordingButton } from './RecordingButton/RecordingButton';

export function Popup(){
  const [isRecording, setIsRecording] = useState(false);

  useEffect(() => {
    const initializeState = async () => {
      const isRecording = await StateService.getRecordingState()
      setIsRecording(isRecording)
    }

    initializeState()
  }, [])

  return (
    <div className="App">

      <Logo isRecording={isRecording}/>
      
      {isRecording &&
        <h1 className="RecordingText">
          Recording
        </h1>
      }

      <RecordingButton
        isRecording={isRecording}
        onClick={() => {
          setIsRecording(!isRecording)
          StateService.changeRecordingState(!isRecording)
        }}
      />
    </div>
  );
};
