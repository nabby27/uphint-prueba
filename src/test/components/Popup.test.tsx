/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect'
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { Popup } from "../../pages/Popup/Popup";

describe('Popup', () => {
    it(`
        WHEN Popup is rendered
        THEN Popup should render correctly
    `, () => {
        const component = shallow(<Popup />);

        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it(`
        WHEN Popup is rendered
        THEN should be unrecorded
    `, () => {
        render(<Popup />);

        screen.getByRole('button', { name: 'Start Recording' });
    });

    it(`
        GIVEN rendered Popup
        WHEN click to Start Recording button
        THEN should be start recording
    `, async () => {
        render(<Popup />);
        const button = screen.getByRole('button', { name: 'Start Recording' });

        await userEvent.click(button);

        screen.getByRole('button', { name: 'Stop Recording' });
        screen.getByRole('heading', {level: 1, name: 'Recording'});
    });

    it(`
        GIVEN Popup is recording
        WHEN click to Stop Recording button
        THEN should be stop recording
    `, async () => {
        render(<Popup />);
        const startRecordingButton = screen.getByRole('button', { name: 'Start Recording' });
        await userEvent.click(startRecordingButton);

        const stopRecordingButton = screen.getByRole('button', { name: 'Stop Recording' });
        await userEvent.click(stopRecordingButton);

        screen.getByRole('button', { name: 'Start Recording' });
        const recordingText = screen.queryByRole('heading', {level: 1, name: 'Recording'});
        expect(recordingText).not.toBeInTheDocument();
    });
})