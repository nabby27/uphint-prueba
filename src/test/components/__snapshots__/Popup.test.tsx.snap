// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup 
        WHEN Popup is rendered
        THEN Popup should render correctly
     1`] = `
<div
  className="App"
>
  <Logo
    isRecording={false}
  />
  <RecordingButton
    isRecording={false}
    onClick={[Function]}
  />
</div>
`;
